contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -2070,3 +2079,242 @@ gen_sp_droprolemember_subcmds(const char *user, const char *member)
 	rewrite_object_refs(stmt);
 	return res;
 }
+
+static void
+ValidateLinkedServerDataSource(char* data_src)
+{
+	/* 
+	 * Only treat fully qualified DNS names (endpoints) or IP address 
+	 * as valid data sources.
+	 * 
+	 * If data source is provided in the form of servername\\instancename, we
+	 * throw an error to suggest use of fully qualified domain name or the IP address
+	 * instead.
+	 */
+	if (strchr(data_src, '\\'))
```

Comment:
Better to add a check for the valid case. If that doesn't match, reject everything else.